### DISCUSSION TITLE
Reading a PathF

### DISCUSSION THREAD

**Discussion by TRSkillman** [dotnet/maui#9852] [General] [answered]
Hi,
I've discovered that Graphics PathF has a host of functions useful for reading and therefore editing a path.
Here is a rough sample;

        public PathF NewPath(PathF pathF)
        {
            PathF newpath = new PathF();
            PointF[] points;
            for(int i = 0; i < pathF.OperationCount; i++) {
                var pathop = pathF.GetSegmentType(i);
                switch (pathop)
                {
                    case PathOperation.Move:
                        points = pathF.GetPointsForSegment(i);
                        newpath.Move(points[0].X, points[0].Y);
                        break;
                    case PathOperation.Line:
                        points = pathF.GetPointsForSegment(i);
                        newpath.LineTo(points[0].X, points[0].Y);
                        break;
                    case PathOperation.Cubic:
                        points = pathF.GetPointsForSegment(i);
                        newpath.CurveTo(points[0], points[1], points[2]);
                        break;
                }
            }
            
            return newpath;
        }

My question is, can I rely on using these functions in the future or will they change?
Thanks

**Comment by jsuarezruiz** [comment]
This is also what we use to get the flattened path, bounding box etc. There are no plans to change, and it will be exactly the same in .NET 7.

**Comment by TRSkillman** [comment, marked as answer]
That's great to know!
Thank you.
